csharp/ql/lib/semmle/code/csharp/Member.qll

```diff
@@ -98,10 +98,21 @@ class Modifiable extends Declaration, @modifiable {
 
   /** Holds if this declaration is `unsafe`. */
   predicate isUnsafe() {
-    this.hasModifier("unsafe") or
-    this.(Parameterizable).getAParameter().getType() instanceof PointerType or
-    this.(Property).getType() instanceof PointerType or
-    this.(Callable).getReturnType() instanceof PointerType
+    this.hasModifier("unsafe")
+    or
+    exists(Type t, Type child |
+      t = this.(Parameterizable).getAParameter().getType() or
+      t = this.(Property).getType() or
+      t = this.(Callable).getReturnType() or
+      t = this.(DelegateType).getReturnType()
+    |
+      child = t.getAChild*() and
```

Comment:
Why is the `getAChild` needed?

Reply:
I think the example on the PR description illustrates why it is needed.

Reply:
Isn't that just a `delegate` whose return type is a function pointer? Wouldn't `this.(DelegateType).getReturnType() instanceof FunctionPointerType` hold there?

Reply:
For now the `getAChild` only has an effect on the actual .NET 7 stubs being created for arrays of pointers.
An example could be
```csharp
public void FromManaged(T*[] array, System.Span<TUnmanagedElement> buffer)
```